src/vector/_backends/object_.py

```diff
@@ -1509,11 +1501,9 @@ def _gather_coordinates(
             raise TypeError("specify t= or tau=, but not more than one")
         temporal = TemporalObjectT(coordinates.pop("t"))
     elif "tau" in coordinates:
-        if "t" in coordinates:
-            raise TypeError("specify t= or tau=, but not more than one")
         temporal = TemporalObjectTau(coordinates.pop("tau"))
 
-    if len(coordinates) == 0:
+    if not coordinates:
```

Comment:
If you know the type of x, `if x` is perfectly valid. I do find Python's truthiness to be handy in some cases, for example, `[x for x in arr or ()]`; this will allow arr to be None (or False). Until and if we get None-coalescing operators, this is a handy method to keep things clean. I would allow truthiness in the follow cases, I think:

1. A boolean type
2. A collection, which could possibly be None (but not if it can be a numpy.array)
3. A string, which could possibly be None
4. _Maybe_ for an int. True and False are equivalent to ints, after all.

The benefit for 2 and 3 is that unlike "len", the truthiness will also protect against None, which can really simpify important type protections. It also is handled correctly by MyPy (`if x` will type-narrow None out).

However, that said, I didn't catch this one (maybe swept in with other changes?) and am just fine with it being written this way. I think both read well (though if perforce matters, len and eq might be a hair slower?). The len one clearly indicates it's an empty collection.

Reply:
I believe Python truthiness is why there's no `.empty()` method in Python. Ruby has `.empty?`, and C++ has `.empty()`.

Reply:
Since there's an `in coordinates` above used on this, this has to work correctly, so I'll leave it as `if not coordinates` unless you have enough of a preference to reply; if it was a non-collection the `in` would break before this. But happy to change it, the other form reads just as well.